### ISSUE TITLE
install_name_tool crashes on OSX 10.8

### ISSUE THREAD

**Issue by zwelchWI** [conda/conda-build#145]
I recently started running into this error message when attempting to build Cyclus for OSX 10.8.  I've tried manually setting -headerpad and -headerpad_max_install_names, but it didn't seem to have any effect.  I monkeyed around with the conda-build source code and simply ignoring the error allows Cyclus to build and run without issues, though I'm not sure this is general enough to be an actual fix.
```
install_name_tool: changing install names or rpaths can't be redone for: /Volumes/MacintoshHD2/condor/slot3/dir_78018/userdir/anaconda/envs/_build/lib/cyclus/stubs/libStubFacility.dylib (for architecture x86_64) because larger updated load commands do not fit (the program must be relinked, and you may need to use -headerpad or -headerpad_max_install_names)
Traceback (most recent call last):
  File "/Volumes/MacintoshHD2/condor/slot3/dir_78018/userdir/anaconda/bin/conda-build", line 5, in <module>
    sys.exit(main())
  File "/Volumes/MacintoshHD2/condor/slot3/dir_78018/userdir/anaconda/lib/python2.7/site-packages/conda_build/main_build.py", line 93, in main
    args.func(args, p)
  File "/Volumes/MacintoshHD2/condor/slot3/dir_78018/userdir/anaconda/lib/python2.7/site-packages/conda_build/main_build.py", line 224, in execute
    build.build(m, verbose=not args.quiet, post=post)
  File "/Volumes/MacintoshHD2/condor/slot3/dir_78018/userdir/anaconda/lib/python2.7/site-packages/conda_build/build.py", line 325, in build
    binary_relocation=bool(m.get_value('build/binary_relocation', True)))
  File "/Volumes/MacintoshHD2/condor/slot3/dir_78018/userdir/anaconda/lib/python2.7/site-packages/conda_build/post.py", line 199, in post_build
    mk_relative(f, binary_relocation=binary_relocation)
  File "/Volumes/MacintoshHD2/condor/slot3/dir_78018/userdir/anaconda/lib/python2.7/site-packages/conda_build/post.py", line 178, in mk_relative
    mk_relative_osx(path)
  File "/Volumes/MacintoshHD2/condor/slot3/dir_78018/userdir/anaconda/lib/python2.7/site-packages/conda_build/post.py", line 146, in mk_relative_osx
    macho.install_name_change(path, osx_ch_link)
  File "/Volumes/MacintoshHD2/condor/slot3/dir_78018/userdir/anaconda/lib/python2.7/site-packages/conda_build/macho.py", line 59, in install_name_change
    subprocess.check_call(args)
  File "/Volumes/MacintoshHD2/condor/slot3/dir_78018/userdir/anaconda/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '[u'install_name_tool', u'-change', u'libboost_system.dylib', u'@loader_path/../../libboost_system.dylib', u'/Volumes/MacintoshHD2/condor/slot3/dir_78018/userdir/anaconda/envs/_build/lib/cyclus/stubs/libStubFacility.dylib']' returned non-zero exit status 1
```

**Comment by ocefpaf** [comment]
@asmeurer any news on how to solve this?  I am experience the same issue when building `NCO` on OSX.

https://travis-ci.org/ioos/conda-recipes/builds/79042660#L3294

**Comment by asmeurer** [comment]
@ocefpaf are you compiling with -headerpad_max_install_names?

**Comment by rsignell-usgs** [comment]
Indeed, I don't see the `-headerpad_max_install_names` on this line here:
https://travis-ci.org/ioos/conda-recipes/builds/79042660#L3293

**Comment by ocefpaf** [comment]
> @ocefpaf are you compiling with -headerpad_max_install_names?

I had a few combinations of that and they all failed like @zwelchWI original issue for Cyclus.

Since my PR is a little old let me try again and I will report back here...

> Indeed, I don't see the -headerpad_max_install_names on this line here:

That is because I revert the PR to a pristine state :wink:

**Comment by asmeurer** [comment]
If that flag doesn't work I'm not sure what else to do. The one time I've seen that error that flag fixed it for me. If anyone figures it out please post the solution here.

**Comment by ocefpaf** [comment]
I got my mistake.  I was passing more arguments to `LDFLAGS` along with
`-headerpad_max_install_names`. (LDFLAGS=$LDFLAGS + opts.)
If I pass only `-headerpad_max_install_names` it works just fine.  Thanks @asmeurer.
